src/lib/utils/build-project-url.ts

```diff
@@ -0,0 +1,50 @@
+// TODO: Temporary types until centralized types for projects are added.
+
+interface GitHubSource {
+  type: 'github';
+  repoName: string;
+  ownerName: string;
+  url: string;
+}
+
+interface GitLabSource {
+  type: 'gitlab';
+  repoName: string;
+  ownerName: string;
+  host: string;
+  url: string;
+}
+
+interface RadicleSource {
+  type: 'radicle';
+  rid: string;
+  repoName: string;
+  seed: string;
+  url: string;
+}
+
+interface GenericGitSource {
+  type: 'generic';
+  repoName: string;
+  url: string;
+}
+
+type Source = GitHubSource | GitLabSource | RadicleSource | GenericGitSource;
+
+/**
+ * Builds a project profile URL from a project source object.
+ * @param source The source to build the URL from.
+ * @returns The URL.
+ */
+export default function (source: Source) {
+  switch (source.type) {
+    case 'github':
+      return `/app/projects/github/${source.ownerName}/${source.repoName}`;
```

Comment:
Minor: how about
const root = '/app/projects/';
return `${root}/${source.type}/...`
so that it's all dynamic?

Reply:
I thought about it, but decided it'd be nicer to have all the different url schemas written out explicitly. If you feel strongly, happy to change.